### DISCUSSION TITLE
Normalization of Fourier transform

### DISCUSSION THREAD

**Discussion by sijieh23** [spectralDNS/shenfun#104] [Q&A]
Hi,

Recently, I've started using `Shenfun`. I'm wondering how the Fourier transform is normalized in `Shenfun`, especially when it comes to multidimensional cases.

Take 2D as an example. Is it divided by the number of grid points, $N_x\cdot N_y$, in the `forward` transform, or is this normalization done in the `backward` transform, as is common in most FFT packages (such as those in `MATLAB`, `numpy`, and `scipy`)?

I've been experimenting with the code, and it seems like `Shenfun` follows the `ortho` strategy as in `scipy`, in which both the `forward` and `backward` transforms are scaled by $1/\sqrt{N_x \cdot N_y}$. However, I'm not sure if this is correct.

**Comment by mikaem** [comment]
Hi
The forward transform in Shenfun in any direction is divided by N. So in 2D with two Fourier spaces it is divided by $N_x \cdot N_y$. Note that a consecutive forward and backward should return the original data. You will probably find the FFTs better described in the [mpi4py_fft](https://github.com/mpi4py/mpi4py-fft) source code.
